fe2-android/app/src/test/framework/robolectric/java/com/github/dedis/popstellar/di/DataRegistryModuleHelper.java

```diff
@@ -16,19 +16,38 @@ public static DataRegistry buildRegistry() {
 
   public static DataRegistry buildRegistry(LAORepository laoRepository, KeyManager keyManager) {
     return buildRegistry(
-        laoRepository, new MessageRepository(), keyManager, new ServerRepository());
+        laoRepository,
+        new SocialMediaRepository(),
+        new MessageRepository(),
+        keyManager,
+        new ServerRepository());
+  }
+
+  public static DataRegistry buildRegistry(
+      LAORepository laoRepo, SocialMediaRepository socialMediaRepo, KeyManager keyManager) {
+    return buildRegistry(
+        laoRepo, socialMediaRepo, new MessageRepository(), keyManager, new ServerRepository());
+  }
+
+  public static DataRegistry buildRegistry(
+      LAORepository laoRepo,
+      MessageRepository msgRepo,
+      KeyManager keyManager,
+      ServerRepository serverRepo) {
+    return buildRegistry(laoRepo, new SocialMediaRepository(), msgRepo, keyManager, serverRepo);
   }
 
   public static DataRegistry buildRegistry(
       LAORepository laoRepo,
+      SocialMediaRepository socialMediaRepo,
       MessageRepository msgRepo,
       KeyManager keyManager,
       ServerRepository serverRepo) {
     LaoHandler laoHandler = new LaoHandler(keyManager, msgRepo, laoRepo, serverRepo);
     RollCallHandler rollCallHandler = new RollCallHandler(keyManager, laoRepo);
     ElectionHandler electionHandler = new ElectionHandler(msgRepo, laoRepo);
     ConsensusHandler consensusHandler = new ConsensusHandler(laoRepo);
-    ChirpHandler chirpHandler = new ChirpHandler(laoRepo);
+    ChirpHandler chirpHandler = new ChirpHandler(laoRepo, socialMediaRepo);
     TransactionCoinHandler transactionCoinHandler = new TransactionCoinHandler(laoRepo);
```

Comment:
I'm a bit concerned that when we'll be done with the decoupling, this static method will have a huge parameter list...

Reply:
That's exactly the concern I raised in #1210. It is better, but far from perfect.

I don't have any good solution right now too.
Maybe a builder with a default value for each repository.